src/common/webrequest_curl.cpp

```diff
@@ -362,217 +510,1194 @@ void wxWebAuthChallengeCURL::SetCredentials(const wxWebCredentials& cred)
+
+    if ( selectStatus < 0 )
+    {
+        // Massive error: report an error on each socket.
+        for ( PollActions::iterator it = m_pollActions.begin() ;
+              it != m_pollActions.end() ; ++it )
+        {
+            wxThreadEvent* event =
+                new wxThreadEvent(wxEVT_SOCKET_POLLER_RESULT);
+            event->SetPayload<wxSOCKET_T>(it->first);
+            event->SetInt(SocketPoller::HAS_ERROR);
+            wxQueueEvent(m_handler, event);
+        }
+    }
+    else if ( selectStatus == 0 )
+    {
+        ;// select timed out.  There is no need to do anything.
+    }
+    else // selectStatus > 0. There was activity on at least 1 socket.
+    {
+        if ( (m_readEnd != -1) && FD_ISSET(m_readEnd, &readFds) )
+        {
+            char c;
+            recv(m_readEnd, &c, 1, 0);
+        }
+
+        for ( PollActions::iterator it = m_pollActions.begin() ;
+              it != m_pollActions.end() ; ++it )
+        {
+            wxSOCKET_T sock = it->first;
+
+            if ( m_disabledList.find(sock) != m_disabledList.end() )
+            {
+                continue;
+            }
+
+            int checkActions = it->second;
+            int result = SocketPoller::INVALID_RESULT;
+
+            if ( checkActions & SocketPoller::POLL_FOR_READ )
+            {
+                if ( FD_ISSET(sock, &readFds) )
+                {
+                    result |= SocketPoller::READY_FOR_READ;
+                }
+            }
+
+            if ( checkActions & SocketPoller::POLL_FOR_WRITE )
+            {
+                if ( FD_ISSET(sock, &writeFds) )
+                {
+                    result |= SocketPoller::READY_FOR_WRITE;
+                }
+            }
+
+            if ( FD_ISSET(sock, &errorFds) )
+            {
+                result |= SocketPoller::HAS_ERROR;
+            }
+
+            if ( result != SocketPoller::INVALID_RESULT )
+            {
+                wxThreadEvent* event =
+                    new wxThreadEvent(wxEVT_SOCKET_POLLER_RESULT);
+                event->SetPayload<wxSOCKET_T>(sock);
+                event->SetInt(result);
+                wxQueueEvent(m_handler, event);
+                m_disabledList.insert(sock);
+            }
+        }
+    }
+}
+
+SocketPollerImpl* SocketPollerImpl::Create(wxEvtHandler* hndlr)
+{
+    return new SelectSocketPoller(hndlr);
+}
+
+#endif
+
+//
+// wxWebSessionCURL
+//
+
+int wxWebSessionCURL::ms_activeSessions = 0;
+unsigned int wxWebSessionCURL::ms_runtimeVersion = 0;
+int wxWebSessionCURL::ms_progressFuncContinue = 0;
+
+wxWebSessionCURL::wxWebSessionCURL() :
+    m_handle(NULL)
+{
+    // Initialize CURL globally if no sessions are active
+    if ( ms_activeSessions == 0 )
+    {
+        if ( curl_global_init(CURL_GLOBAL_ALL) )
+        {
+            wxLogError(_("libcurl could not be initialized"));
+            wxLogError(_("libcurl could not be initialized"));
```

Comment:
```suggestion
```

Reply:
Not sure how this happened. It's been removed.